pkg/micropython/include/micropython.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2019 Inria
+ *               2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_micropython
+ *
+ * @{
+ *
+ * @file
+ * @brief       MicroPython RIOT specific API
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef MICROPYTHON_H
+#define MICROPYTHON_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef MP_RIOT_HEAPSIZE
+#define MP_RIOT_HEAPSIZE (8U*1024)
+#endif
+
+/**
+ * @brief   Initialize RIOT MicroPython port
+ *
+ * @param[in]   heap        ptr to heap MicroPython should use
+ * @param[in]   heap_size   size of heap
+ */
+void mp_riot_init(char* heap, size_t heap_size);
+
+/**
+ * @brief   Execute a string as MicroPython code
+ *
+ * @param[in]   src pointer to Python code
+ * @param[in]   len length of src
+ */
+void mp_do_str(const char *src, int len);
```

Comment:
Do strings get executed in the same environment/interpreter sessions, or is it a new session each time?

Reply:
I've added a note (there's one global instance and the code will be executed in that instance).